sstables/sstables.cc

```diff
@@ -1415,10 +1416,13 @@ void sstable::maybe_rebuild_filter_from_index(uint64_t num_partitions, sstring o
         return;
     }
 
+    auto old_filter_ptr = dynamic_cast<utils::filter::bloom_filter*>(_components->filter.get());
+    assert(old_filter_ptr);
+
```

Comment:
Isn't it better to have on_internal_error here? Crash in test, throw (and maybe recover) in production.

Reply:
I leave the decision to you. It's a tradeoff between potentially losing more availability than needed, and potentially continuing to run the program in an uncertain state (and not getting a coredump from it). What we value more is arbitrary.

Do you want an `on_internal_error` here?

Reply:
I think there's a good chance the system will recover. At least, the operator will be able to take the node down and restart it with fresh schema. So I do want an on_internal_error here.

It reduces our ability to find the bug quickly, but also reduces the chance of a coordinated crash in multiple nodes.

Reply:
Done. Take a look at commit `sstables: for i_filter downcasts...` again.